libs/barista-components/experimental/combobox/src/combobox.ts

```diff
@@ -352,8 +352,12 @@ export class DtCombobox<T>
       .pipe(takeUntil(this._destroy))
       .subscribe(() => {
         this._autocomplete._additionalOptions = this._options.toArray();
-        this._setSelectionByValue(this._value, this._initialOptionChange);
+        Promise.resolve().then(() => {
```

Comment:
NIT: would add a comment that this is done to prevent an expression changed after checked error.